File: food_project/food_project/models/MenuItem.js
const mongoose = require('mongoose');

const MenuItemSchema = new mongoose.Schema({
  name: { type: String, required: true },
  price: { type: Number, required: true },
  imageUrl: { type: String },          // cloud URL (optional)
  imageBase64: { type: String },       // base64 data URL string (data:image/..;base64,...)
  description: { type: String },
  available: { type: Boolean, default: true },
  soldCount: { type: Number, default: 0 }
}, { timestamps: true });

module.exports = mongoose.model('MenuItem', MenuItemSchema);

File: food_project/food_project/flutter_food_app/lib/models/menu_item.dart
class MenuItemModel {
  String id;
  String name;
  double price;
  String imageUrl;
  String? imageBase64;
  String description;
  bool available;

  MenuItemModel({required this.id, required this.name, required this.price, required this.imageUrl, this.imageBase64, this.description = '', this.available = true});

  factory MenuItemModel.fromJson(Map<String, dynamic> json) => MenuItemModel(
    id: json['_id'],
    name: json['name'],
    price: (json['price'] as num).toDouble(),
    imageUrl: json['imageUrl'] ?? '',
    imageBase64: json['imageBase64'],
    description: json['description'] ?? '',
    available: json['available'] ?? true
  );
}
